raft/src/main/java/org/apache/kafka/raft/SimpleKeyValueStore.java

```diff
@@ -140,7 +144,7 @@ private Records buildRecords(Map<K, V> map) {
         for (Map.Entry<K, V> entry : map.entrySet()) {
             records[i++] = serialize(entry.getKey(), entry.getValue());
         }
-        return MemoryRecords.withRecords(CompressionType.NONE, records);
+        return MemoryRecords.withRecords(currentPosition.offset, CompressionType.NONE, currentPosition.epoch, records);
```

Comment:
Not sure the point of this change. We should let the leader decide the offset and epoch.